addons/mail/models/discuss/discuss_channel.py

```diff
@@ -633,6 +633,26 @@ def _message_post_after_hook(self, message, msg_vals):
         self._set_last_seen_message(message)
         return super()._message_post_after_hook(message, msg_vals)
 
+    def _process_attachments_for_post(self, attachments, attachment_ids, message_values):
+        return_values = super()._process_attachments_for_post(attachments, attachment_ids, message_values)
+
+        if attachments:
+            voice_metadata_vals = []
+            for attachment, m2m_command in zip(attachments, return_values.get('attachment_ids', [])):
+                if len(attachment) == 3:
+                    _, _, info = attachment
+                    voice = info and info.get('voice')
+                    if voice:
+                        # Extract attachment_id from the (4, attachment_id) M2M command
+                        attachment_id = m2m_command[1] if m2m_command[0] == 4 else None
+                        if attachment_id:
+                            voice_metadata_vals.append({
+                                'attachment_id': attachment_id,
+                            })
+            if voice_metadata_vals:
+                self.env['discuss.voice.metadata'].create(voice_metadata_vals)
+        return return_values
```

Comment:
Hello,

Isn't it supposed to be done in `_post_add_create`?

The voice kwargs might not be correctly passed (see recent fix https://github.com/odoo/odoo/pull/184506)
or the method might not be called in the flow you are trying to fix?

Reply:
Hello @seb-odoo 

The `_post_add_create` isn't triggered in the flow.

Maybe we can do,

```py
# Extract attachment_id from the (4, attachment_id) M2M command
attachment_id = m2m_command[1] if m2m_command[0] == 4 else None
if attachment_id:
    self.env['ir.attachment'].browse(attachment_id).exists()._post_add_create(voice=True)
```

What say?

Reply:
If you look at the docstring: "Overrides behaviour when the attachment is created through the controller"